pkg/provider/fake/fake.go

```diff
@@ -124,16 +127,47 @@ func (p *Provider) PushSecret(_ context.Context, value []byte, _ corev1.SecretTy
 	return nil
 }
 
-// Empty GetAllSecrets.
-func (p *Provider) GetAllSecrets(_ context.Context, _ esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
-	// TO be implemented
-	return nil, fmt.Errorf("GetAllSecrets not implemented")
+// GetAllSecrets returns multiple secrets from the given ExternalSecretFind
+// Currently, only the Name operator is supported.
+func (p *Provider) GetAllSecrets(_ context.Context, ref esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
+	if ref.Name != nil {
+		matcher, err := find.New(*ref.Name)
+		if err != nil {
+			return nil, err
+		}
+
+		latestDataMap := make(map[string]*Data)
+		for key, data := range p.config {
```

Comment:
Okay I do have one last one thought. 😂 you can use the same for loop to set up both maps since you are just overriding the keys if I under this correctly. 😁

Reply:
Thanks, fixed it!

Reply:
Noice! :)